python/lsst/daf/butler/script/exportCalibs.py

```diff
@@ -0,0 +1,152 @@
+        calibrations.
+    collections : `list` [`str`]
+       Data collections to pull calibrations from.  Must be an
+       existing `~CollectionType.CHAINED` or
+       `~CollectionType.CALIBRATION` collection.
+
+    Returns
+    -------
+    datasetTable : `astropy.table.Table`
+        A table containing relevant information about the calibrations
+        exported.
+
+    Raises
+    ------
+    RuntimeError :
+        Raised if the output directory already exists.
+    """
+    butler = Butler(repo, writeable=False)
+
+    calibTypes = [datasetType for datasetType in butler.registry.queryDatasetTypes(...)
+                  if datasetType.isCalibration()]
+
+    collectionsToExport = []
+    datasetsToExport = []
+
+    for collection in collections:
+        log.info("Checking collection: %s", collection)
+
+        # Get collection information.
+        collectionsToExport.append(collection)
+        collectionType = butler.registry.getCollectionType(collection)
+        if collectionType == CollectionType.CHAINED:
+            # Iterate over the chain:
+            collectionsToExport.append(collection)
+            collectionRecords = butler.registry.getCollectionChain(collection)
+
+            for child in collectionRecords:
+                childType = butler.registry.getCollectionType(child)
+                collectionsToExport.append(child)
+                if childType == CollectionType.CALIBRATION:
+                    exportCollections, exportDatasets = parseCalibrationCollection(butler.registry,
+                                                                                   child,
+                                                                                   calibTypes)
+                    collectionsToExport.extend(exportCollections)
+                    datasetsToExport.extend(exportDatasets)
+        elif collectionType == CollectionType.CALIBRATION:
+            exportCollections, exportDatasets = parseCalibrationCollection(butler.registry,
+                                                                           collection,
+                                                                           calibTypes)
+            collectionsToExport.append(collection)
+            collectionsToExport.extend(exportCollections)
+            datasetsToExport.extend(exportDatasets)
+
+    if os.path.exists(directory):
+        raise RuntimeError(f"Export directory exists: {directory}")
+    os.makedirs(directory)
+    with butler.export(directory=directory, format="yaml", transfer="auto") as export:
+        collectionsToExport = list(set(collectionsToExport))
+        datasetsToExport = list(set(datasetsToExport))
+
+        for exportable in collectionsToExport:
+            try:
+                export.saveCollection(exportable)
+            except Exception as e:
+                log.warning("Did not save collection %s due to %s.", exportable, e)
+
+        try:
+            log.info("Saving dataset(s)")
```

Comment:
Can you say you are saving N datasets (it's already a list so counting them is not expensive).